addons/account/models/account_move.py

```diff
@@ -288,6 +288,18 @@ def _get_default_invoice_incoterm(self):
     inalterable_hash = fields.Char(string="Inalterability Hash", readonly=True, copy=False)
     string_to_hash = fields.Char(compute='_compute_string_to_hash', readonly=True)
 
+    _sql_constraints = [
+        # Partial constraint, complemented by unique index (see below). Still
+        # useful to keep because it provides a proper error message when a
+        # violation occurs, as it shares the same prefix as the unique index.
+        ('unique_name', 'unique (id)', 'Posted Journal Entries\' numbers must be unique by journal.'),
+    ]
+
+    def init(self):
+        self.env.cr.execute('SELECT indexname FROM pg_indexes WHERE indexname = %s', ('account_move_unique_name_index',))
+        if not self.env.cr.fetchone():
+            self.env.cr.execute("CREATE UNIQUE INDEX account_move_unique_name_index ON account_move (name, journal_id) WHERE name != '/';")
```

Comment:
On second thought, are we sure about the `name != '/'` condition? It seems that it could be annoying when renumbering entries in a journal, as even entries that have been reverted to draft will conflict between themselves. Perhaps the condition `state = 'posted'` is better in this sense?

Reply:
Yes, it would also be aligned with the previous constraint :+1: